clients/src/main/java/org/apache/kafka/server/authorizer/AclCreateResult.java

```diff
@@ -35,10 +37,10 @@ public AclCreateResult(ApiException exception) {
     }
 
     /**
-     * Returns any exception during create. If exception is null, the request has succeeded.
+     * Returns any exception during create. If exception is empty, the request has succeeded.
      */
-    public ApiException exception() {
-        return exception;
+    public Optional<ApiException> exception() {
```

Comment:
Is `failed()` below still useful enough?

Reply:
@ijuma Thanks for the review. Agree `failed()` is unnecessary, removed.